### PR TITLE
chore(EMA) #28402 : SDK: Support Vanity URLs in Page API

### LINKED ISSUE #28402: SDK: Support Vanity URLs in Page API
### Parent
#28401

### User Story
As a dotCMS user, I want the Page API to support vanity URLs, so that when a vanity URL is associated with a page and the status code is a redirect (301 or 302), the API will return the content of the vanity URL instead of the regular page object.

```[tasklist]
### Acceptance Criteria
- [x] Support vanity URLs and return the page object normally if the status code is a forward 200.
- [x] Return the content of the vanity URL if the status code is a redirect 301 or 302.
- [x] Add minimal Vanity URL data when processing a `Forward` code, and add the `X-DOT-VanityUrl` Response Header as well.
```

### Proposed Objective
Core Features

### Proposed Priority
Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.:
N/A

### Assumptions & Initiation Needs
- Ensure that the vanity URLs are properly configured in the dotCMS system.
- Verify that the status codes for the vanity URLs are set correctly.

### Quality Assurance Notes & Workarounds
- QA should verify that the Page API returns the correct content based on the vanity URL status code.
- Workaround: Manually check the API response to ensure the correct content is returned for vanity

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
### Proposed Changes
* Adding a minified version of the Vanity URL object to the JSON response when processing a Forward status code, for troubleshooting purposes.
* Adding the `X-DOT-VanityUrl` Response Header just like we do when accessing Vanity URLs in the front-end.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java [hunk 1]
```diff
@@ -325,6 +327,7 @@ private Response getPageRender(final HttpServletRequest originalRequest,
         final Optional<CachedVanityUrl> cachedVanityUrlOpt =
                 this.pageResourceHelper.resolveVanityUrlIfPresent(originalRequest, uri, languageId);
         if (cachedVanityUrlOpt.isPresent()) {
+            response.setHeader(VanityUrlAPI.VANITY_URL_RESPONSE_HEADER, cachedVanityUrlOpt.get().vanityUrlId);
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java:330]: cool

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/VanityURLView.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+/**
+ * This View represents the main information of a Vanity URL in dotCMS.
+ *
+ * @author Jose Castro
+ * @since May 8th, 2024
+ */
+public class VanityURLView implements Serializable {
+
+    private final String id;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/VanityURLView.java:15]: nice

**Review by jdotcms — APPROVED** [review]
Nice work Chepe

**Comment by dsilvam** [comment]
Any opportunity to include automated testing here @jcastro-dotcms? Postman?

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/PageContext.java [hunk 3]
```diff
@@ -42,14 +47,16 @@ public PageContext(
             final PageMode pageMode,
             final HTMLPageAsset page,
             final boolean graphQL,
-            final boolean parseJSON) {
+            final boolean parseJSON,
+            final VanityURLView vanityUrl) {
```
[the remark this answers is not in the document]
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/PageContext.java:51]: @dsilvam there is already a builder for it: `com.dotmarketing.portlets.htmlpageasset.business.render.PageContextBuilder`

**Review by dsilvam — APPROVED** [review]

**Pull request merged**